core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -62,4 +64,12 @@ private[history] abstract class ApplicationHistoryProvider {
    */
   def getConfig(): Map[String, String] = Map()
 
+  /**
+   * Get the [[Path]]s to the Event log files. For legacy event log directories, directory path
+   * itself is returned. The caller is responsible for listing the files and using them as needed.
+   * If the attemptId is [[None]], event logs corresponding to all attempts for the given
+   * application are downloaded as a single zip file.
+   */
+  def getEventLogPaths(appId: String, attemptId: Option[String]): Seq[Path] = Seq.empty
```

Comment:
Just a minor issue for now, but having `Path` here kinda assumes that the logs exist in a file system, which may not be the case for different provider implementations.